tests/syslog/test_syslog.py

```diff
@@ -10,67 +10,67 @@
-
-    # Add Rsyslog destination for testing
-    duthost.shell("sudo config syslog add {}".format(local_syslog_srv_ip))
-    logger.debug("Added new rsyslog server IP {}".format(local_syslog_srv_ip))
-
-    yield
-
-    # Remove the syslog configuration
-    duthost.shell("sudo config syslog del {}".format(local_syslog_srv_ip))
-
-
-def test_syslog(duthosts, enum_rand_one_per_hwsku_frontend_hostname, ptfhost, config_dut):
+@pytest.mark.parametrize("dummy_syslog_server_ip_a, dummy_syslog_server_ip_b", [("10.0.80.166", None), ("fd82:b34f:cc99::100", None), ("10.0.80.165", "10.0.80.166"), ("fd82:b34f:cc99::100", "10.0.80.166"), ("fd82:b34f:cc99::100", "fd82:b34f:cc99::200")])
+def test_syslog(duthosts, enum_rand_one_per_hwsku_frontend_hostname, dummy_syslog_server_ip_a, dummy_syslog_server_ip_b):
     duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
     logger.info("Starting syslog tests")
     test_message = "Basic Test Message"
 
-    logger.debug("Configuring rsyslog server")
-    config_syslog_srv(ptfhost)
+    logger.info("Configuring the DUT")
+    # Add dummy rsyslog destination for testing
+    if dummy_syslog_server_ip_a is not None:
+        duthost.shell("sudo config syslog add {}".format(dummy_syslog_server_ip_a))
+        logger.debug("Added new rsyslog server IP {}".format(dummy_syslog_server_ip_a))
+    if dummy_syslog_server_ip_b is not None:
+        duthost.shell("sudo config syslog add {}".format(dummy_syslog_server_ip_b))
+        logger.debug("Added new rsyslog server IP {}".format(dummy_syslog_server_ip_b))
+
+    logger.info("Start tcpdump")
+    tcpdump_task, tcpdump_result = duthost.shell("sudo timeout 20 tcpdump -i any -s0 -A -w {} \"udp and port 514\"".format(DUT_PCAP_FILEPATH), module_async=True)
+    # wait for starting tcpdump
+    time.sleep(5)
 
     logger.debug("Generating log message from DUT")
     # Generate a syslog from the DUT
     duthost.shell("logger --priority INFO {}".format(test_message))
 
-    # Check syslog messages for the test message
-    def _check_syslog():
-        result = ptfhost.shell("grep {} /var/log/syslog | grep -v ansible | grep -c \"{}\""
-                               .format(duthost.hostname, test_message))["stdout"]
-        return result != "0"
+    # wait for stoping tcpdump 
+    tcpdump_task.close()
+    tcpdump_task.join()
 
-    pytest_assert(wait_until(SYSLOG_MESSAGE_TEST_TIMEOUT, 1, _check_syslog),
-                  "Test syslog message not seen on the server after {}s".format(SYSLOG_MESSAGE_TEST_TIMEOUT))
+    # Remove the syslog configuration
+    if dummy_syslog_server_ip_a is not None:
+        duthost.shell("sudo config syslog del {}".format(dummy_syslog_server_ip_a))
+    if dummy_syslog_server_ip_b is not None:
+        duthost.shell("sudo config syslog del {}".format(dummy_syslog_server_ip_b))
+
+    duthost.fetch(src=DUT_PCAP_FILEPATH, dest=DOCKER_TMP_PATH)
+
+    # Check pcap file for the destination IPs
+    def _check_pcap(dummy_ip_a, dummy_ip_b):
```

Comment:
Suggest to define this `_check_pcap` file in the scope of this script instead of this test case.